src/main/scala/com/mwz/sonar/scala/scoverage/ScoverageSensor.scala

```diff
@@ -0,0 +1,160 @@
+/*
+ * Sonar Scala Plugin
+ * Copyright (C) 2018 All contributors
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+package com.mwz.sonar.scala.scoverage
+
+import com.mwz.sonar.scala.Scala
+import com.mwz.sonar.scala.util.JavaOptionals._
+import org.sonar.api.batch.fs.{FileSystem, InputComponent, InputFile}
+import org.sonar.api.batch.sensor.{Sensor, SensorContext, SensorDescriptor}
+import org.sonar.api.config.Configuration;
+import org.sonar.api.utils.log.Loggers
+import scala.collection.JavaConverters._
+import scala.util.{Failure, Success, Try}
+
+/**
+ *  Main sensor for importing Scoverage reports to SonarQube.
+ *
+ *  @author BalmungSan
+ */
+final class ScoverageSensor extends ScoverageSensorInternal with ScoverageReportParser
+
+/** Implementation of the sensor */
+private[scoverage] class ScoverageSensorInternal extends Sensor {
+  //cake pattern to mock the scoverage report parser in tests
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Space after the start of the comment [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.sonar-scala.com/coding_rules#rule_key=Scalastyle%3Ascalastyle_SpaceAfterCommentStartChecker)